src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+const roundDown = (num: number): number => Math.floor(num * 100) / 100;
+
+export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    const max = Math.max(a, b, c);
 
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+      || max >= a + b + c - max
+    ) {
+      throw new Error('sizes are wrong!');
+    }
+  }
+
+  getArea(): number {
+    const { a, b, c } = this;
+
+    const semiPerimetr = (a + b + c) / 2;
+
+    return roundDown(Math.sqrt(semiPerimetr
+      * (semiPerimetr - a)
+      * (semiPerimetr - b)
+      * (semiPerimetr - c)));
+  }
 }
 
 export class Circle {
+  shape = Shape.Circle;
 
+  constructor(
+    public color: Color,
+    private radius: number,
+  ) {
+    if (this.radius <= 0) {
+      throw new Error('Wrong radius!');
+    }
```

Comment:
you check input values for figure sizes in each constructor
will be better to create separate function for this purpose and call this function in each class